crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -158,4 +158,108 @@ module std::vector {
         v.swap(i, last_idx);
         v.pop_back()
     }
+
+    // === Macros ===
+
+    /// Destroy the vector `v` by calling `f` on each element and then destroying the vector.
+    public macro fun destroy<$T>($v: vector<$T>, $f: |$T|) {
+        let mut v = $v;
+        while (!v.is_empty()) {
+            $f(v.pop_back());
+        };
+        v.destroy_empty();
+    }
+
+    /// Map the vector `v` to a new vector by applying the function `f` to each element.
+    public macro fun map<$T, $U>($v: vector<$T>, $f: |$T| -> $U): vector<$U> {
```

Comment:
It may be worthwhile to reverse first, as many people expect map to operate in-order. If not, at least clarify that order is unspecified in the doc comment.

Reply:
Just submitted a follow up with `do` and `do_mut`!